client/src/utils/FetchUtil.ts

```diff
@@ -1,7 +1,34 @@
 import { DataFile, DataSource } from '@dekkai/data-source';
 import s3Client from '@/services/S3Service';
 
+const storeSizeLimit = 104857600; // 100MB in bytes
+const storeEntryLimit = 10485760; // 10MB in bytes
 const memoizedStore = new Map();
+const memoizedSize = new Map();
+
+const storeSize = (): number => {
+  let totalValue = 0;
+  for (const value of memoizedSize.values()) {
+    totalValue += value;
+  }
+  return totalValue;
+};
+
+const storeResult = (hash: string, result: unknown, size: number): void => {
+  if (size > storeEntryLimit) {
+    return;
+  }
+
+  // TO-DO: Optimize this
+  while (storeSize() + size > storeSizeLimit) {
+    const shiftKey = memoizedStore.keys().next().value;
```

Comment:
I think we should remove the term `memoized` here. Memoization implies storing the result of a previously run function, however, this store is used mostly to store a copy of data that's origin lies on another machine.

Cache is probably the more general and well understood term.